iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/ProjectContext.java

```diff
@@ -19,14 +19,35 @@
  */
 package org.sonar.iac.kubernetes.visitors;
 
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.sonar.iac.kubernetes.model.ProjectResource;
+
 /**
  * Data class to provide information about the project. This allows to share cross-file knowledge to the individual checks.
  */
 public final class ProjectContext {
+  private final Map<String, Map<String, Set<ProjectResource>>> projectResourcePerNamespacePerPath = new HashMap<>();
 
   private ProjectContext() {
   }
 
+  public Set<ProjectResource> getProjectResource(String namespace, String path, Class<? extends ProjectResource> clazz) {
+    if (projectResourcePerNamespacePerPath.containsKey(namespace)) {
+      var resourcesPerPath = projectResourcePerNamespacePerPath.get(namespace);
+      if (resourcesPerPath.containsKey(path)) {
```

Comment:
I'm not sure about this part. I remember that we should add some logic to traverse the pathes, not only checking for equals.

Reply:
After checking with Rudy, there's still some things to do here. Will write via slack.